bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/message/MsgFactory.java

```diff
@@ -93,44 +108,48 @@ public void addData(byte[] data, int len) {
         // Now see if we have enough data for a complete message.
         // If not, we return null, and expect this method to be called again
         // when more data has come in.
-        int msgLen = -1;
-        boolean isExtended = false;
         if (end > 1) {
             // we have some data, but do we have enough to read the entire header?
             int headerLength = Msg.getHeaderLength(buf[1]);
-            isExtended = Msg.isExtended(buf, end, headerLength);
+            boolean isExtended = Msg.isExtended(buf, end, headerLength);
             logger.trace("header length expected: {} extended: {}", headerLength, isExtended);
             if (headerLength < 0) {
                 removeFromBuffer(1); // get rid of the leading 0x02 so draining works
-                bail("got unknown command code " + Utils.getHexByte(buf[1]));
+                bail("got unknown command code " + Utils.getHexByte(buf[0]));
             } else if (headerLength >= 2) {
                 if (end >= headerLength) {
                     // only when the header is complete do we know that isExtended is correct!
-                    msgLen = Msg.getMessageLength(buf[1], isExtended);
+                    int msgLen = Msg.getMessageLength(buf[1], isExtended);
+                    logger.trace("msgLen expected: {}", msgLen);
                     if (msgLen < 0) {
                         // Cannot make sense out of the combined command code & isExtended flag.
                         removeFromBuffer(1);
-                        bail("unknown command code/ext flag: " + Utils.getHexByte(buf[1]));
+                        bail("got unknown command code/ext flag " + Utils.getHexByte(buf[0]));
+                    } else if (msgLen > 0) {
+                        if (end >= msgLen) {
+                            msg = Msg.createMessage(buf, msgLen, isExtended);
+                            removeFromBuffer(msgLen);
+                        }
+                    } else { // should never happen
+                        logger.warn("invalid message length, internal error!");
                     }
                 }
             } else { // should never happen
                 logger.warn("invalid header length, internal error!");
-                msgLen = -1;
             }
         }
-        logger.trace("msgLen expected: {}", msgLen);
-        Msg msg = null;
-        if (msgLen > 0 && end >= msgLen) {
-            msg = Msg.createMessage(buf, msgLen, isExtended);
-            removeFromBuffer(msgLen);
+        // indicate no more messages available in buffer if empty or undefined message
+        if (end == 0 || msg == null) {
+            logger.trace("done processing current buffer data");
+            done = true;
         }
         logger.trace("keeping buffer len {} data: {}", end, Utils.getHexString(buf, end));
```

Comment:
```suggestion
        if(logger.isTraceEnabled()){
            logger.trace("keeping buffer len {} data: {}", end, Utils.getHexString(buf, end));
        }
```

Reply:
There are a ton of `logger.trace` calls across the code. What's the point of this?